packages/nextcloud/README.md

```diff
@@ -25,15 +22,46 @@ dependency_overrides:
 ```
 You can either remove the `ref` or use a commit hash. It's not recommended to remove it, because then the version will be updated very often.
 
-## Development
+## Usage
+
+### Authentication
 
-Except for WebDAV all client code is generated using OpenAPI specs which can be found in the `../../specs/` folder.  
-Templates for these OpenAPI specs are generated from the Nextcloud codebase to make development easier.  
+There are multiple ways to authenticate.  
+First there is HTTP Basic auth which works with the normal user credentials (e-mail and other identifiers also work):
+```dart
+final client = NextcloudClient(
+    Uri.parse('http://localhost'),
+    loginName: 'admin',
+    password: 'admin',
+);
+```
+
+Secondly there is Http Bearer auth which works with app passwords:
+```dart
+final client = NextcloudClient(
+    Uri.parse('http://localhost'),
+    loginName: 'admin',
+    appPassword: 'xxxxx-xxxxx-xxxxx-xxxx-xxxxx',
+);
+```
 
-To generate such a template take a look at `../../tool/generate-nextcloud.sh`.  
-After you have generated a template, you need to fill it out. Some endpoints can or have to be discarded.  
+Not all endpoints work with just HTTP Basic auth, so it is advised to use app passwords obtained either directly in the Web UI by the user or using the [login flow](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2).  
+Some endpoints do not need any authentication at all or provided extended information when the request is authenticated.
```

Comment:
This sounds off. I guess you mean (but I'm not sure):
```suggestion
Some endpoints do not need any authentication at all or provide extended information when the request needs to be authenticated.
```

Reply:
`is` is correct. The request can be optionally authenticated and in that case it will return more information.